### ISSUE TITLE
Wrong results with non-contiguous arrays

### ISSUE THREAD

**Issue by dsuess** [uber/neuropod#384]
## Bug

Neuropod doesn't deal correctly with non-contiguous numpy arrays as outputs. 

## To Reproduce

1. Create `module` directory, copy the following code into `module/export.py`:

```
import click
import numpy as np
from pathlib import Path


def split(x):
    x1 = x[:, :2]
    x2 = x[:, 2:]
    return {'x1': x1, 'x2': x2}


def split_contiguous(x):
    x1 = np.ascontiguousarray(x[:, :2])
    x2 = np.ascontiguousarray(x[:, 2:])
    return {'x1': x1, 'x2': x2}


def get_model(data_path):
    return split


@click.group()
def main():
    pass


@main.command('export')
def export():
    from neuropod.packagers import create_python_neuropod
    x = np.arange(16).reshape(4, 4)
    print(x.dtype)

    create_python_neuropod(
        neuropod_path='test.np',
        model_name='splitter',
        data_paths=[],
        entrypoint_package='module.export',
        entrypoint='get_model',
        code_path_spec=[
            {"python_root": str(Path(__file__).parents[1].absolute()), "dirs_to_package": ["module"]}
        ],
        input_spec=[{'name': 'x', 'dtype': 'int64', "shape": (4, 4)}],
        output_spec=[
            {'name': 'x1', 'dtype': 'int64', "shape": (4, 2)},
            {'name': 'x2', 'dtype': 'int64', "shape": (4, 2)}
        ],
        test_input_data={'x': x},
        test_expected_out={'x1': x[:, :2], 'x2': x[:, 2:]}
    )


@main.command('run')
def run():
    from neuropod import loader
    x = np.arange(16).reshape(4, 4)

    with loader.load_neuropod('test.np') as pod:
        result = pod.infer({'x': x})

    np.testing.assert_array_equal(result['x1'], x[:, :2])
    np.testing.assert_array_equal(result['x2'], x[:, 2:])
    print('Done')


if __name__ == '__main__':
    main()
```

2. Export the neuropod using `python module/export.py export`. It's written to `test.np`
3. Run the test using `python module/export.py run` -- the assertion fails.
4. In `get_model`, replace `return split` by `return split_contiguous` and run steps 2 & 3 again -- the assertions pass

## Expected behavior

Assertions in `run` should not fail. 

## Environment

 - Neuropod Version (e.g., 0.2.0): `0.2.0`
 - OS (Linux, macOS): Linux
 - Language (Python, C++, Go bindings): Python
 - Python version: 3.7.4
 - Using OPE: no
- Numpy version: `1.18.1`
 
If this bug report is about running a specific model: N/A (pure numpy model)

If running on GPU: N/A
 
 - Any other relevant information:

**Comment by dsuess** [comment]
That TODO does look pretty much on point :D

Maybe it's worth just putting in a error for now if the data is not contiguous. It took me a while to figure out because the testcase run in `create_python_neuropod` passes. If a user is alerted when the neuropod is created, it's pretty easy to fix.

**Comment by VivekPanyam** [comment]
Put up a fix in #399. Thanks again for the detailed bug report and repro!
